src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -60,9 +60,11 @@ public void EndStreamValueScope()
         /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
         public async Task<Stream> StartStreamValueScopeAsync()
         {
-            this.WriteSeparatorIfNecessary();
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            this.WriteItemWithSeparatorIfNeeded();
             this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
-            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+            await this.FlushAsync().ConfigureAwait(false);

```

Comment:
Why the change?

Reply:
I made this change in order to be consistent with `StartStreamValueScope` (the sync version). I would say the reasoning behind forcing a flush here is to ensure that the buffer is empty before we proceed with writing to the stream, with the expectation that writing to a stream may often involve writing large chunks of data. If the next value they write is smaller than the chunk size, then we'll attempt to write it all without buffering, which might lead to our bufferWriter being resizied (i.e. extra allocations). So the forced flush is just to reduce the chances of extending the internal buffer size.